Automattic/wp-calypso
# Simple payments block: don't attempt to save product without user capabilities

Stops attempts to publish product record if the user does not have capabilities.

This started as an attempt to implement some kind of notification about not being able to save products as a contributor level user, similar to what we have in Calypso: https://github.com/Automattic/wp-calypso/pull/27109

Instead of showing a warning badge and disabling the form, user _can_ edit the form, values are stored in attributes and all works smoothly. Flow fully described in https://github.com/Automattic/wp-calypso/issues/28661#issuecomment-440051606 . 

We just try to avoid saving the product when it shouldn't work. 

I couldn't find a Gutenberg way to check for capabilities — we could manually do a `apiFetch` but since core is using this method to render "publish" button, I'd be okay with doing the same until better capabilities-check support is in Gutenberg.

Closes #28661

#### Changes proposed in this Pull Request
- Check if current post has "publish" action
- If action is missing, don't attempt to save product record
- Form values are stored in attributes and the block is created when 

#### Testing instructions
- Create a post as "contributor" level user
- Insert simple payments block
- Save post for review
- Observe no errors in console, no network requests to create a product.
    Without the change you would see these errors:
    <img width="734" alt="image" src="https://user-images.githubusercontent.com/87168/48737070-b81bb900-ec55-11e8-9bfe-ea768c8c1da0.png">

    If you see `Unhandled promise rejection Error: "[object Object]"` error, that's expected as it seems to be an issue with core and shows up even without this block.
- Open the same post as admin user and publish the post; observe network call for creating the product
